ProjFS.Mac/PrjFSKext/KauthHandler.cpp

```diff
@@ -47,6 +48,359 @@ struct PendingRenameOperation
     thread_t thread;
 };
 
+class NullTracer
+{
+    NullTracer() = delete;
+    NullTracer(const NullTracer&) = delete;
+    NullTracer& operator=(const NullTracer&) = delete;
+public:
+    explicit NullTracer(kauth_action_t vnodeAction, vnode_t vnode)
+    {}
+    explicit NullTracer(kauth_action_t fileopAction, uintptr_t arg0, uintptr_t arg1, uintptr_t arg2)
+    {}
+    
+    void Flush()
+    {
+    }
+    void Printf(const char* format, ...) __attribute__ ((format (printf, 2, 3)))
+    {
+    }
+    void SendProviderMessage(MessageType providerMessage)
+    {}
+    void SendProviderMessageResult(bool success)
+    {}
+    
+    void SetDeniedForCrawler()
+    {}
+
+    void SetVnodeOpResult(int result)
+    {
+    }
+};
+
+static const char* MessageTypeString(MessageType messageType)
```

Comment:
should this live in KextLog?

Reply:
Yes, or perhaps `Message_Kernel.cpp`. Plenty of code still needs rearranging in this PR; I didn't want to spend any time on aesthetics until I knew I was building the right thing.